### PR TITLE
Added v2->v3 migration notes. Piggybacked some minor fixes.

### PR DESCRIPTION
**Pull request by zajko**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This release candidate is compatible with [#node RC2](https://github.com/casper-
 
 - `CLAccountHash` now has `toFormattedStr()` and `fromFormattedStr(hexStr: string)` methods
 - Changed `enum KeyVariant` to `enum KeyTag`, backfilled all the variants that were missing, added condor-specifin ones
-- `StoredValue` has new fields which represent enum variants and are reuqired for 2.x compatibility:
+- `StoredValue` has new fields which represent enum variants and are reuqired for `condor` compatibility:
```
Comment (ACStone-MTS) [inline CHANGELOG.md]:
```suggestion
- `StoredValue` has new fields which represent `enum` variants and are required for `Condor` compatibility:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ This release candidate is compatible with [#node RC2](https://github.com/casper-
 - `merkleProof` field to `EraSummary` class
 - optional `latest_switch_block_hash` added to `GetStatusResult`
 - field `execution_info` of type `ExecutionInfo` was added. This affects `CasperServiceByJsonRPC.getDeployInfo` method return types.
-- `block_with_signatures` was added of type `JsonBlockWithSignatures` was added. `JsonBlockWithSignatures.block` is a union type of either `Version1` or `Version2`. `Version1` variant should be returned for blocks cretaed prior to 2.x migration, `Version2` for blocks created after 2.x migration. This change affects `CasperServiceByJsonRPC.getBlockInfo` method return type.
+- `block_with_signatures` was added of type `JsonBlockWithSignatures` was added. `JsonBlockWithSignatures.block` is a union type of either `Version1` or `Version2`. `Version1` variant should be returned for blocks created prior to `condor` migration, `Version2` for blocks created after `condor` migration. This change affects `CasperServiceByJsonRPC.getBlockInfo` method return type.
```
Comment (ACStone-MTS) [inline CHANGELOG.md]:
```suggestion
- `block_with_signatures` with type `JsonBlockWithSignatures` was added. `JsonBlockWithSignatures.block` is a union type of either `Version1` or `Version2`. `Version1` variant should be returned for blocks created prior to `Condor` migration, `Version2` for blocks created after `Condor` migration. This change affects the `CasperServiceByJsonRPC.getBlockInfo` method return type.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -51,15 +51,16 @@ This release candidate is compatible with [#node RC2](https://github.com/casper-
 - `delegators` entry field from `Bid` class changed structure
 - `CasperServiceByJsonRPC.getEraInfoBySwitchBlock` argument changed from `string` to `BlockIdentifier`
 - `CasperServiceByJsonRPC.getEraSummary` argument changed from `string` to `BlockIdentifier`
-- `ExecutionResult` class changed to a type union of either `Version1` or `Version2`. For deploys created pre-2.x `ExecutionResult` should contain `Version1`, for post `2.x` they will be `Version2` variants.
+- `ExecutionResult` class changed to a type union of either `Version1` or `Version2`. For deploys created pre-condor `ExecutionResult` should contain `Version1`, for post `condor` they will be `Version2` variants.
```
Comment (ACStone-MTS) [inline CHANGELOG.md]:
```suggestion
- `ExecutionResult` class changed to a type union of either `Version1` or `Version2`. For deploys created pre-Condor `ExecutionResult` should contain `Version1`, for post `Condor` they will be `Version2` variants.
```

File: resources/migration-guide_v2-v3.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
+# v2 to v3 migration guide
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
# V2 to V3 Migration Guide
```

File: resources/migration-guide_v2-v3.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+The `v3` release made several breaking changes to inputs, outputs that were present in `v2`. All of them are attempts to follow how the API of the node changed. `V3` does not introduce any changes to the behavior of the sdk outside of those that were forced by the changes in the node. This document will explain changes on an endpoint by endpoint basis.
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
The `v3` release makes several breaking changes to inputs and outputs that were present in `v2`. All changes to the SDK are driven by changes to the API of the node. `V3` does not introduce any changes to the behavior of the SDK, outside of those that were forced by the changes in the node. This document will explain all changes on an endpoint by endpoint basis.
```

File: resources/migration-guide_v2-v3.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+## Breaking changes
+
+### getAccountInfo
+
+This util function will be usable only to fetch unmigrated data from the node. All legacy `Account` data in the noe will be gradually migrated to the new `AddressableEntity` format. This is a lazy process and will happen once someone deploys a `Transaction` using an `Account`. To fetch accounts created after node 2.x migration or ones that were migrated to the new format, use `CasperServiceByJsonRPC.getEntity`.
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
This `util` function will be usable only to fetch unmigrated data from the node. All legacy `Account` data in the node will be gradually migrated to the new `AddressableEntity` format. This is a "lazy" process and will happen once someone deploys a `Transaction` using an `Account`. To fetch accounts created after node 2.x migration or ones that were migrated to the new format, use `CasperServiceByJsonRPC.getEntity`.
```

File: resources/migration-guide_v2-v3.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+### CasperServiceByJsonRPC
+
+#### getDeployInfo
+
+The output data for this endpoint changed. The `GetDeployResult` used to have `deploy` and `execution_results` fields. `deploy` should not have changed, but the `execution_results` field was removed. It was replaced by optional `execution_info` field of `ExecutionInfo` type. `ExecutionInfo` contains `execution_result` (singular, not an array as previously) of `ExecutionResult` type. `ExecutionResult` type changed to be a union of either `ExecutionResultV1` or `ExecutionResultV2`. `ExecutionResultV1` corresponds to `ExecutionResult` prior to the discussed changes.
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
The output data for this endpoint changed. The `GetDeployResult` previously contained `deploy` and `execution_results` fields. `deploy` should not have changed, but the `execution_results` field was removed. It was replaced by optional `execution_info` field of `ExecutionInfo` type. `ExecutionInfo` contains `execution_result` (singular, not an array as previously) of `ExecutionResult` type. `ExecutionResult` type changed to be a union of either `ExecutionResultV1` or `ExecutionResultV2`. `ExecutionResultV1` corresponds to `ExecutionResult` prior to the discussed changes.
```

File: resources/migration-guide_v2-v3.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+#### getBlockInfo
+
+This method change it's output data (`GetBlockResult`). Currently it returns a structure with one field `block_with_signatures` which is of type `JsonBlockWithSignatures`. `JsonBlockWithSignatures.signatures` are the signatures of the block, `JsonBlockWithSignatures.block` is a union type that represents the block data. It should be of the structure:
+
+```
+{ Version1: BlockV1 }
+```
+
+for blocks that were produced prior to `condor` migration, and
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
for blocks that were produced prior to `Condor` migration, and
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -51,15 +51,16 @@ This release candidate is compatible with [#node RC2](https://github.com/casper-
...
+- `getAccountInfo` function will only work before `condor` node release
```
Comment (ACStone-MTS) [inline CHANGELOG.md]:
```suggestion
- `getAccountInfo` function will only work before `Condor` node release
```

File: resources/migration-guide_v2-v3.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+```
+{ Version2: BlockV2 }
+```
+
+for blocks produced after the migration. `BlockV1` is the same as type `JsonBlock` in `v2` of casper-js-sdk.
+
+#### getBlockInfoByHeight
+
+All the changes that were made to output data of `getBlockInfo` were also made to `getBlockInfoByHeight`.
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
All changes made to output data of `getBlockInfo` were also made to `getBlockInfoByHeight`.
```

File: resources/migration-guide_v2-v3.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+#### getLatestBlockInfo
+
+All the changes that were made to output data of `getBlockInfo` were also made to `getLatestBlockInfo`.
```
Comment (ACStone-MTS) [inline resources/migration-guide_v2-v3.md]:
```suggestion
All changes made to output data of `getBlockInfo` were also made to `getLatestBlockInfo`.
```

**Review by ACStone-MTS — COMMENTED** [review]
Capitalizing "Condor" is at your discretion. I don't think it's a problem to leave as is, just hits my eye weird.

**Review by hoffmannjan — APPROVED** [review]
Awesome job!

**Pull request merged**